src/components/Navbar/Navbar.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState } from "react";
+import { IntlProvider } from "react-intl";
+import { BrowserRouter } from "react-router-dom";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { messages } from "../../App/App.const";
+import Navbar from "./Navbar";
+import {
+  DarkModeContext,
+  DarkModeContextProvider,
+} from "../../contexts/DarkModeContext";
+
+describe("testing navbar component", () => {
+  function NavbarMock() {
+    const [search, setSearch] = useState("");
+    const [language, setLanguage] = useState("en");
```

Comment:
do takich rzeczy wykorzystuje się mocki - czyli takie fikcyjne funkcje, które mają za zadanie zasymulować funkcje odpalane w komponencie:
https://profil-software.com/blog/development/testing-react-application-react-hooks-jest-and-enzyme-newbies/

Reply:
https://gist.github.com/mauricedb/eb2bae5592e3ddc64fa965cde4afe7bc

Reply:
https://www.linkedin.com/pulse/mocking-react-hooks-usestate-useeffect-leonard-lin/